pkg/binary/envoy/debug/admin_test.go

```diff
@@ -51,3 +53,18 @@ func TestEnableEnvoyAdminDataCollection(t *testing.T) {
 		require.NotEmpty(t, f.Size(), "file %v was empty", path)
 	}
 }
+
+// runAndTerminateWithDebug is like RequireRunTerminate, except returns a directory populated by the debug plugin.
+func runAndTerminateWithDebug(t *testing.T, workingDir string, debug func(r *envoy.Runtime) error, args ...string) error {
+	fakeEnvoy, removeFakeEnvoy := morerequire.RequireCaptureScript(t, "envoy")
+	defer removeFakeEnvoy()
+
+	o := &globals.RunOpts{EnvoyPath: fakeEnvoy, WorkingDir: workingDir, DontArchiveWorkingDir: true}
+
+	r := envoy.NewRuntime(o)
+	r.Out = io.Discard
```

Comment:
we don't check envoy output in the debug utilities, only the process state collection